repos/spack_repo/builtin/packages/erf/package.py

```diff
@@ -0,0 +1,161 @@
+        depends_on("amrex")
+        for v in ("mpi", "openmp", "cuda", "particles"):
+            depends_on(f"amrex+{v}", when=f"+{v}")
+            depends_on(f"amrex~{v}", when=f"~{v}")
+        for sm in CudaPackage.cuda_arch_values:
+            depends_on(f"amrex+cuda cuda_arch={sm}", when=f"+cuda cuda_arch={sm}")
+        depends_on("mpi", when="+mpi")
+        depends_on("cuda@11.0:", when="+cuda")
+        depends_on("fftw", when="+fft")
+
+        with when("+netcdf"):
+            depends_on("netcdf-c+mpi+parallel-netcdf", when="+mpi")
+            depends_on("netcdf-c~mpi+parallel-netcdf", when="~mpi")
+            depends_on("netcdf-fortran")
+            depends_on("hdf5+mpi", when="+mpi")
+            depends_on("hdf5~mpi", when="~mpi")
+
+    conflicts("+openmp", when="+cuda", msg="Cannot enable both OpenMP and CUDA")
+    conflicts("+fft", when="~mpi", msg="FFT support requires MPI")
+    conflicts("+radiation", when="platform=darwin", msg="Radiation is not supported on macOS")
+
+    _KOKKOS_SM_TO_FLAG = {
+        "70": "Kokkos_ARCH_VOLTA70",
+        "72": "Kokkos_ARCH_VOLTA72",
+        "75": "Kokkos_ARCH_TURING75",
+        "80": "Kokkos_ARCH_AMPERE80",
+        "86": "Kokkos_ARCH_AMPERE86",
+        "89": "Kokkos_ARCH_ADA89",
+        "90": "Kokkos_ARCH_HOPPER90",
+    }
+
+    def _kokkos_flags_from_cuda_arch(self, sm_list):
+        for sm in sm_list:
+            flag = self._KOKKOS_SM_TO_FLAG.get(sm)
+            if flag:
+                yield flag
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("ERF_ENABLE_MPI", "mpi"),
+            self.define_from_variant("ERF_ENABLE_OMP", "openmp"),
+            self.define_from_variant("ERF_ENABLE_NETCDF", "netcdf"),
+            self.define_from_variant("ERF_ENABLE_PARTICLES", "particles"),
+            self.define_from_variant("ERF_ENABLE_MULTIBLOCK", "multiblock"),
+            self.define_from_variant("ERF_ENABLE_RADIATION", "radiation"),
+            self.define_from_variant("ERF_BUILD_TESTS", "tests"),
+            self.define_from_variant("ERF_BUILD_FCOMPARE", "fcompare"),
+            self.define_from_variant("ERF_ENABLE_FFT", "fft"),
+            self.define("ERF_DIM", "3"),
+        ]
+        args.append(self.define("ERF_USE_EXTERNAL_AMREX", True))
+        args.append(self.define("ERF_CLONE_AMREX", False))
+        args.append(self.define("GIT_SUBMODULE_PROTOCOL", "https"))
+        args.append(self.define("MPIEXEC_PREFLAGS", "--oversubscribe"))
+        args.append(self.define("ERF_DIM", "3"))
+
+        if "+netcdf" in self.spec:
+            args.extend(
+                [
+                    self.define("NetCDF_C_PATH", self.spec["netcdf-c"].prefix),
+                    self.define("NetCDF_FORTRAN_PATH", self.spec["netcdf-fortran"].prefix),
+                ]
+            )
+
+        if "+radiation" in self.spec:
+            args.extend(
+                [
+                    self.define("ERF_ENABLE_KOKKOS", True),
```

Comment:
This seems to imply that erf needs a `depends_on("kokkos", when="+radiation")` added above